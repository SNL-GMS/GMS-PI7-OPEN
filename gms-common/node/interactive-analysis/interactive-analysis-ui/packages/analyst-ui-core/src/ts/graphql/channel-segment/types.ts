
export enum ChannelSegmentType {
    ACQUIRED = 'ACQUIRED',
    RAW = 'RAW',
    DETECTION_BEAM = 'DETECTION_BEAM',
    FK_BEAM = 'FK_BEAM',
    FILTER = 'FILTER'
}

export enum TimeSeriesType {
    WAVEFORM = 'WAVEFORM',
    FK_SPECTRA = 'FK_SPECTRA',
    DETECTION_FEATURE_MAP = 'DETECTION_FEATURE_MAP'
}

export interface TimeSeries {
    startTime: number;
    sampleRate: number;
    sampleCount: number;
}

export interface ChannelSegment<T extends TimeSeries> {
    id: string;
    channelId: string;
    name?: string;
    type: ChannelSegmentType;
    timeseriesType: TimeSeriesType;
    startTime: number;
    endTime: number;
    timeseries: T[];
  }
