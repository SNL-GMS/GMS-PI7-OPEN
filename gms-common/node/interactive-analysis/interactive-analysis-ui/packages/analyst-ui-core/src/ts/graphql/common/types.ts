// ***************************************
// Mutations
// ***************************************

// ***************************************
// Subscriptions
// ***************************************

// ***************************************
// Queries
// ***************************************

// ***************************************
// Model
// ***************************************

/**
 * Creation Type, reflects system change or analyst change
 */
export enum CreatorType {
  Analyst = 'Analyst',
  System = 'System'
}

/**
 * Distance value's units degrees or kilometers
 */
export enum DistanceUnits {
  degrees = 'degrees',
  km = 'km'
}

/**
 * Distance to source type
 */
export enum DistanceSourceType {
  Event = 'Event',
  UserDefined = 'UserDefined'
}

/**
 * Time range in epoch seconds
 */
export interface TimeRange {
  startTime: number;
  endTime: number;
}

/**
 * Location information
 */
export interface Location {
  latDegrees: number;
  lonDegrees: number;
  elevationKm: number;
}

/**
 * Position information relative to a location
 */
export interface Position {
  northDisplacementKm: number;
  eastDisplacementKm: number;
  verticalDisplacementKm: number;
}

/**
 * Log Level to determine different levels
 */
export enum LogLevel {
  INFO = 'INFO',
  DEBUG = 'DEBUG',
  WARNING = 'WARNING',
  ERROR = 'ERROR',
  DATA = 'DATA'
}

/**
 * Client Log Input
 */
export interface ClientLogInput {
  logLevel: LogLevel;
  message: string;
  time?: string;
}

/**
 * Client log mutation arguments
 */
export interface ClientLogMutationArgs {
  input: ClientLogInput;
}

/**
 * Creation Info
 */
export interface CreationInfo {
  id: string;
  creationTime: number;
  creatorId: string;
  creatorType: CreatorType;
  creatorName: string;
}

/**
 * Units used in DoubleValue part of feature prediction
 */
export enum Units {
    DEGREES = 'DEGREES',
    SECONDS= 'SECONDS',
    SECONDS_PER_DEGREE = 'SECONDS_PER_DEGREE',
    UNITLESS = 'UNITLESS'
}

/**
 * ENUM list of phase types
 */
export enum PhaseType {
  P = 'P',
  S = 'S',
  P3KPbc = 'P3KPbc',
  P4KPdf_B = 'P4KPdf_B',
  P7KPbc = 'P7KPbc',
  P7KPdf_D = 'P7KPdf_D',
  PKiKP = 'PKiKP',
  PKKSab = 'PKKSab',
  PKP2bc = 'PKP2bc',
  PKP3df_B = 'PKP3df_B',
  PKSab = 'PKSab',
  PP_1 = 'PP_1',
  pPKPbc = 'pPKPbc',
  PS = 'PS',
  Rg = 'Rg',
  SKiKP = 'SKiKP',
  SKKSac = 'SKKSac',
  SKPdf = 'SKPdf',
  SKSdf = 'SKSdf',
  sPdiff = 'sPdiff',
  SS = 'SS',
  sSKSdf = 'sSKSdf',
  Lg = 'Lg',
  P3KPbc_B = 'P3KPbc_B',
  P5KPbc = 'P5KPbc',
  P7KPbc_B = 'P7KPbc_B',
  Pb = 'Pb',
  PKKP = 'PKKP',
  PKKSbc = 'PKKSbc',
  PKP2df = 'PKP2df',
  PKPab = 'PKPab',
  PKSbc = 'PKSbc',
  PP_B = 'PP_B',
  pPKPdf = 'pPKPdf',
  PS_1 = 'PS_1',
  SKKP = 'SKKP',
  SKKSac_B = 'SKKSac_B',
  SKS = 'SKS',
  SKSSKS = 'SKSSKS',
  sPKiKP = 'sPKiKP',
  SS_1 = 'SS_1',
  SSS = 'SSS',
  nNL = 'nNL',
  P3KPdf = 'P3KPdf',
  P5KPbc_B = 'P5KPbc_B',
  P7KPbc_C = 'P7KPbc_C',
  PcP = 'PcP',
  PKKPab = 'PKKPab',
  PKKSdf = 'PKKSdf',
  PKP3 = 'PKP3',
  PKPbc = 'PKPbc',
  PKSdf = 'PKSdf',
  pPdiff = 'pPdiff',
  PPP = 'PPP',
  pSdiff = 'pSdiff',
  Sb = 'Sb',
  SKKPab = 'SKKPab',
  SKKSdf = 'SKKSdf',
  SKS2 = 'SKS2',
  Sn = 'Sn',
  sPKP = 'sPKP',
  SS_B = 'SS_B',
  SSS_B = 'SSS_B',
  NP = 'NP',
  P3KPdf_B = 'P3KPdf_B',
  P5KPdf = 'P5KPdf',
  P7KPdf = 'P7KPdf',
  PcS = 'PcS',
  PKKPbc = 'PKKPbc',
  PKP = 'PKP',
  PKP3ab = 'PKP3ab',
  PKPdf = 'PKPdf',
  Pn = 'Pn',
  pPKiKP = 'pPKiKP',
  PPP_B = 'PPP_B',
  pSKS = 'pSKS',
  ScP = 'ScP',
  SKKPbc = 'SKKPbc',
  SKP = 'SKP',
  SKS2ac = 'SKS2ac',
  SnSn = 'SnSn',
  sPKPab = 'sPKPab',
  sSdiff = 'sSdiff',
  NP_1 = 'NP_1',
  P4KPbc = 'P4KPbc',
  P5KPdf_B = 'P5KPdf_B',
  P7KPdf_B = 'P7KPdf_B',
  Pdiff = 'Pdiff',
  PKKPdf = 'PKKPdf',
  PKP2 = 'PKP2',
  PKP3bc = 'PKP3bc',
  PKPPKP = 'PKPPKP',
  PnPn = 'PnPn',
  pPKP = 'pPKP',
  PPS = 'PPS',
  pSKSac = 'pSKSac',
  ScS = 'ScS',
  SKKPdf = 'SKKPdf',
  SKPab = 'SKPab',
  SKS2df = 'SKS2df',
  SP = 'SP',
  sPKPbc = 'sPKPbc',
  sSKS = 'sSKS',
  P4KPdf = 'P4KPdf',
  P5KPdf_C = 'P5KPdf_C',
  P7KPdf_C = 'P7KPdf_C',
  Pg = 'Pg',
  PKKS = 'PKKS',
  PKP2ab = 'PKP2ab',
  PKP3df = 'PKP3df',
  PKS = 'PKS',
  PP = 'PP',
  pPKPab = 'pPKPab',
  PPS_B = 'PPS_B',
  pSKSdf = 'pSKSdf',
  Sdiff = 'Sdiff',
  SKKS = 'SKKS',
  SKPbc = 'SKPbc',
  SKSac = 'SKSac',
  SP_1 = 'SP_1',
  sPKPdf = 'sPKPdf',
  sSKSac = 'sSKSac',
  Sx = 'Sx',
  tx = 'tx',
  N = 'N',
  Px = 'Px',
  PKhKP = 'PKhKP',
  UNKNOWN = 'UNKNOWN'
}
