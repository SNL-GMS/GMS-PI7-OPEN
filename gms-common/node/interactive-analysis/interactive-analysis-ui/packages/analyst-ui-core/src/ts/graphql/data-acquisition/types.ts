import { GraphqlQueryControls } from 'react-apollo';
import { StationType } from '~graphql/station/types';

// ***************************************
// Mutations
// ***************************************

export interface SaveReferenceStationMutationArgs {
  input: ReferenceStation;
}

/**
 * Represents a Reference Station
 */
export interface ReferenceStation {
  name: string;
  description: string;
  stationType: StationType;
  comment: string;
  source: InformationSource;
  latitude: number;
  longitude: number;
  elevation: number;
  actualChangeTime: string;
  systemChangeTime: string;
  aliases: ReferenceAlias[];
}

/**
 * Represents an InformationSource
 */
export interface InformationSource {
  originatingOrganization: string;
  informationTime: string;
  reference: string;
}

/**
 * Represents a ReferenceAlias
 */
export interface ReferenceAlias {
  id: string;
  name: string;
  status: StatusType;
  comment: string;
  actualChangeTime: string;
  systemChangeTime: string;
}

/**
 * Status Types
 */
export enum StatusType {
  Inactive = 'INACTIVE',
  Active = 'ACTIVE'
}

// ***************************************
// Subscriptions
// ***************************************

// ***************************************
// Queries
// ***************************************

// tslint:disable-next-line:max-line-length interface-over-type-literal
export type  TransferredFilesByTimeRangeQueryProps =  {  transferredFilesByTimeRangeQuery: GraphqlQueryControls<{}> & {transferredFilesByTimeRange: FileGap[]}};

/** Arguments for Transferred Files By Time Range Query */
export interface TransferredFilesByTimeRangeQueryArgs {
  timeRange: {
    startTime: number;
    endTime: number;
  };
}

// ***************************************
// Model
// ***************************************

/**
 * Represents the transfer gap fields, current place holder
 * may be moved or renamed in future
 */
export interface FileGap {
  stationName: string;
  channelNames: string[];
  startTime: string;
  endTime: string;
  duration: string;
  location: string;
  priority: string;
}

export interface StatusConfiguration {
  modified: boolean;
}

export interface StationInformation {
  modified: boolean;
}
