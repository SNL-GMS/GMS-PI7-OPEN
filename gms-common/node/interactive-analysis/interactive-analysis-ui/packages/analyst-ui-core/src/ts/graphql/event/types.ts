import { GraphqlQueryControls } from 'react-apollo';
import { FeatureMeasurementValue, SignalDetection } from '~graphql/signal-detection/types';
import { SignalDetectionTypes } from '..';
import { CreationInfo, PhaseType } from '../common/types';

// ***************************************
// Mutations
// ***************************************

export interface AssociationChange {
  events: Event[];
  sds: SignalDetection[];
}
export interface UpdateEventsInput {
  creatorId: string;
  processingStageId: string;
  status?:
    | 'ReadyForRefinement'
    | 'OpenForRefinement'
    | 'AwaitingReview'
    | 'Complete';
  preferredHypothesisId?: string;
  activeAnalystUserNames?: string[];
}

export interface UpdateEventsMutationArgs {
  eventIds: string[];
  input: UpdateEventsInput;
}
export interface UpdateFeaturePredictionsMutationArgs {
  eventId: string;
}

/***
 * Locate Event Mutation Args
 */
export interface LocateEventMutationArgs {
  eventHypothesisId: string;
  preferredLocationSolutionId: string;
  locationBehaviors: LocationBehavior[];
}
export interface ChangeSignalDetectionAssociationsMutationArgs {
  eventHypothesisId: string;
  signalDetectionHypoIds: string[];
  associate: boolean;
}
export interface CreateEventMutationArgs {
  signalDetectionHypoIds: string[];
}

// ***************************************
// Subscriptions
// ***************************************

export interface EventUpdatedSubscription {
  eventsUpdated: {
      id: string;
      status: string;
      activeAnalysts: {
          userName: string;
      }[];
  }[];
}

export interface EventsCreatedSubscription {
  eventsCreated: Event[];
}

// ***************************************
// Queries
// ***************************************

export interface EventsInTimeRangeQueryArgs {
  timeRange: {
    startTime: number;
    endTime: number;
  };
}

// tslint:disable-next-line:max-line-length interface-over-type-literal
export type EventsInTimeRangeQueryProps =  { eventsInTimeRangeQuery: GraphqlQueryControls<{}> & {eventsInTimeRange: Event[]}};

export interface EventsByIdQueryArgs {
  eventId: string;
}

// tslint:disable-next-line:max-line-length interface-over-type-literal
export type EventsByIdQueryProps =  { eventsByIdQuery: GraphqlQueryControls<{}> & {eventsById: Event[]}};

// ***************************************
// Model
// ***************************************

export enum EventStatus {
  ReadyForRefinement = 'ReadyForRefinement',
  OpenForRefinement = 'OpenForRefinement',
  AwaitingReview = 'AwaitingReview',
  Complete = 'Complete',
}

export interface SignalDetectionEventAssociation {
  id: string;
  rejected: boolean;
  signalDetectionHypothesis: {
    id: string;
    rejected: boolean;
    creationInfo: CreationInfo;
  };
  eventHypothesisId: string;
}

export interface EventHypothesis {
  id: string;
  rejected: boolean;
  event: {
    id: string;
    status: EventStatus;
  };
  associationsMaxArrivalTime: number;
  signalDetectionAssociations: SignalDetectionEventAssociation[];
  // Not in OSD, only has a single set and is called locationSolutions
  locationSolutionSets: LocationSolutionSet[];
  preferredLocationSolution: PreferredLocationSolution;
}
export interface Ellipse {
  scalingFactorType: ScalingFactorType;
  kWeight: number;
  confidenceLevel: number;
  majorAxisLength: string;
  majorAxisTrend: number;
  minorAxisLength: string;
  minorAxisTrend: number;
  depthUncertainty: number;
  timeUncertainty: string;
}

export interface Ellipsoid {
  scalingFactorType: ScalingFactorType;
  kWeight: number;
  confidenceLevel: number;
  majorAxisLength: string;
  majorAxisTrend: number;
  majorAxisPlunge: number;
  intermediateAxisLength: number;
  intermediateAxisTrend: number;
  intermediateAxisPlunge: number;
  minorAxisLength: string;
  minorAxisTrend: number;
  minorAxisPlunge: number;
  depthUncertainty: number;
  timeUncertainty: string;
}

/**
 * Represents a preference relationship between an event hypothesis and a location solution.
 * Creation information is included in order to capture provenance of the preference.
 */
export interface PreferredLocationSolution {
  locationSolution: LocationSolution;
  creationInfo: CreationInfo; // in OSD: DNE
}
/**
 * Location Uncertainty for event Location Solution
 */
export interface LocationUncertainty {
  xy: number;
  xz: number;
  xt: number;
  yy: number;
  yz: number;
  yt: number;
  zz: number;
  zt: number;
  tt: number;
  stDevOneObservation: number;
  ellipses: Ellipse [];
  ellipsoids: Ellipsoid [];
}

/**
 * Represents an estimate of the location of an event, defined as latitude, longitude, depth, and time.
 * A location solution is often determined by a location algorithm that minimizes the difference between
 * feature measurements (usually arrival time, azimuth, and slowness) and corresponding feature predictions.
 */
export interface LocationSolution {
  id: string;
  locationType: string;
  location: EventLocation;
  featurePredictions: FeaturePrediction[];
  locationRestraint: LocationRestraint;
  locationUncertainty?: LocationUncertainty;
  locationBehaviors: LocationBehavior[];
  snapshots: SignalDetectionSnapshot[];
}

/**
 * Location Solution Set
 * defines a list of location solutions for an event hypotheis
 * including a snapshot of association when solutions were created
 */
export interface LocationSolutionSet {
  id: string;
  count: number;
  locationSolutions: LocationSolution[];
}

/**
 * Snapshot of state of associations when location solution was created
 */
export interface SignalDetectionSnapshot {
  signalDetectionId: string;
  signalDetectionHypothesisId: string;
  stationId: string;
  stationName: string;
  channelName: string;
  phase: PhaseType;
  time: EventSignalDetectionAssociationValues;
  slowness: EventSignalDetectionAssociationValues;
  azimuth: EventSignalDetectionAssociationValues;
}

/**
 * Generic interface for snapshot values of a signal detection association
 */
export interface EventSignalDetectionAssociationValues {
  defining: boolean;
  observed: number;
  residual: number;
  correction: number;
}
/**
 * Location Behavior for event Location Solution
 */
export interface LocationBehavior {
  residual: number;
  weight: number;
  defining: boolean;
  featurePredictionId: string;
  featureMeasurementId: string;
}

/**
 * Location Restraint for event Location Solution
 */
export interface LocationRestraint {
    depthRestraintType: DepthRestraintType;
    depthRestraintKm: number;
    latitudeRestraintType: RestraintType;
    latitudeRestraintDegrees?: number;
    longitudeRestraintType: RestraintType;
    longitudeRestraintDegrees?: number;
    timeRestraintType: RestraintType;
    timeRestraint?: string;
}
/**
 * 
 * Event Location definition
 */
export interface EventLocation {
  latitudeDegrees: number;
  longitudeDegrees: number;
  depthKm: number;
  time: number;
}
/**
 * Feature Prediction - part of preferred event hypothesis
 */
export interface FeaturePrediction {
  id: string;
  predictedValue: FeatureMeasurementValue;
  predictionType: SignalDetectionTypes.FeatureMeasurementTypeName;
  phase: string;
  channelId?: string;
  stationId?: string;
}

/**
 * Double Value used in OSD common objects
 */
export interface DoubleValue {
  value: number;
  standardDeviation: number;
  // units: Units;
}
export interface PreferredEventHypothesis {
  processingStage: {
    id: string;
  };
  eventHypothesis: EventHypothesis;
}

export interface Event {
  id: string;
  status: EventStatus;
  modified: boolean;
  hasConflict: boolean;
  currentEventHypothesis: PreferredEventHypothesis;
  activeAnalysts: {
    userName: string;
  }[];
  conflictingSdHypIds: string[];
}

/**
 * RestraintType for Location Restraint
 */
export enum RestraintType {
  UNRESTRAINED = 'UNRESTRAINED',
  FIXED = 'FIXED'
}

/**
 * DepthRestraintType for Location Restraint
 */
export enum DepthRestraintType {
  UNRESTRAINED = 'UNRESTRAINED',
  FIXED_AT_DEPTH = 'FIXED_AT_DEPTH',
  FIXED_AT_SURFACE = 'FIXED_AT_SURFACE'
}
/**
 * ScalingFactorType in  Ellipse anbd Ellipsoid
 */
export enum ScalingFactorType {
  CONFIDENCE = 'CONFIDENCE',
  COVERAGE = 'COVERAGE',
  K_WEIGHTED = 'K_WEIGHTED'
}
