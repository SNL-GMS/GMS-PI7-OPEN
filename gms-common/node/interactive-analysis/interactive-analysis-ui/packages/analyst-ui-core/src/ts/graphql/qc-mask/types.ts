import { GraphqlQueryControls } from 'react-apollo';
import { CreationInfo, TimeRange } from '../common/types';

// ***************************************
// Mutations
// ***************************************

export interface QcMaskInput {
  timeRange: TimeRange;
  category: string;
  type: string;
  rationale: string;
}

export interface CreateQcMaskMutationArgs {
  channelIds: string[];
  input: QcMaskInput;
}

export interface RejectQcMaskMutationArgs {
  maskId: string;
  inputRationale: string;
}

export interface UpdateQcMaskMutationArgs {
  maskId: string;
  input: QcMaskInput;
}

// ***************************************
// Subscriptions
// ***************************************

export interface QcMasksCreatedSubscription {
  qcMasksCreated: QcMask[];
}

// ***************************************
// Queries
// ***************************************

export interface QcMasksByChannelIdQueryArgs {
  timeRange: TimeRange;
  channelIds: string[];
}

// tslint:disable-next-line:max-line-length interface-over-type-literal
export type QcMasksByChannelIdQueryProps =  { qcMasksByChannelIdQuery: GraphqlQueryControls<{}> & {qcMasksByChannelId: QcMask[]}};

// ***************************************
// Model
// ***************************************

export interface QcMaskVersion {
  startTime: number;
  endTime: number;
  category: string;
  type: string;
  rationale: string;
  version: string;
  channelSegmentIds: string[];
  creationInfo: CreationInfo;
}

export interface QcMask {
  id: string;
  channelId: string;
  currentVersion: QcMaskVersion;
  qcMaskVersions: QcMaskVersion[];
}
