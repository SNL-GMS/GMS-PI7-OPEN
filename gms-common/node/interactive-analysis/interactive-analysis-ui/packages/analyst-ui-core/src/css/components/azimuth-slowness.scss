@import "~@blueprintjs/core/lib/scss/variables.scss";
.fk-thumbnail {
    outline: 1px solid $gray3;
    position: relative;
    margin-left: 0.22rem;
    margin-right: 0.22rem;
    margin-bottom: 0.44rem;
    &:hover {
        outline: 1px solid $red2;
        cursor: pointer;
    }
    &.selected {
        outline: 3px solid $red3;
    }
    .fk-thumbnail__label {
        color: $black;
        background-color: $gold5;
        position: absolute;
        top: 0px;
        left: 0px;
        padding: 0.15rem;
        z-index: 1;
        font-size: 0.6rem;
    }
    .fk-thumbnail__label--reviewed {
        background-color: $dark-gray3;
        position: absolute;
        top: 0px;
        left: 0px;
        padding: 0.15rem;
        z-index: 1;
        font-size: 0.6rem;
    }
    &:hover .fk-thumbnail__label {
        background-color: $red3;
    }
}
.fk-thumbnail__canvas {
    position: absolute;
    left: 0px;
    right: 0px;
    top: 0px;
    bottom: 0px;
    z-index: 0
}
.azimuth-slowness-data-display {
    width: 100%;
    overflow-y: auto;
    padding: .5rem;
}
.azimuth-slowness-data-display__wrapper {
    flex: 0 0 auto;
    display: flex;
    margin-bottom: 8px;
}
.azimuth-slowness-divider {
    position: relative;
    width: 0.5rem;
    cursor: col-resize;
    flex: 0 0 auto;

}
.azimuth-slowness-divider:hover {
    div {
        background-color: $forest3;
    }
}
.azimuth-slowness-divider__spacer {
    position: absolute;
    left: 0.2rem;
    right: 0.2rem;
    top: 0px;
    bottom: 0px;
}

.fk-axis path,
.fk-axis line {
    fill: none;
    stroke: #ddd;
    shape-rendering: crispEdges;
}

.fk-axis text {
    font-family: sans-serif;
    font-size: 11px;
    fill: #ddd;
}

.fk-weavess-wrapper {
    position: absolute;
    top: 1px;
    bottom: 1px;
    left: 1px;
    right: 4px;
    min-height: 250px;
    color: $gray4;
}
.azimuth-slowness-container {
    padding: 0.5rem;
    height: 100%;
    width: 100%;
    overflow: hidden;
    display: flex;
    user-select: none;
}
.azimuth-slowness-thumbnails {
    flex: 0 0 auto;
    display: flex;
    flex-direction: column;
    overflow: hidden;
}
.azimuth-slowness-thumbnails__control-container {
    position: relative;
    margin-bottom: 0.5rem;
    height: auto;
}
.azimuth-slowness-thumbnails__wrapper-1 {
    flex: 1 1 auto;
    position: relative;
    height: auto;
    overflow: auto;
}
.azimuth-slowness-thumbnails__wrapper-2 {
    width: 100%;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    margin: auto;
}
.azimuth-slowness-thumbnails__wrapper-3 {
    padding: 0.25rem;
    display: flex;
    flex-wrap: wrap;
    align-content: flex-start;
}
.azimuth-slowness-thumbnails-controls__wrapper {
    display: inline-flex;
    flex-direction: row;
    flex: 0 1 auto;
}
.azimuth-slowness-thumbnails-controls {
    display: inline-flex;
    flex-direction: row;
    flex: 0 1 auto;
    align-items: baseline;
}
.azimuth-slowness-thumbnails-controls__label {
    padding-left: 10px;
    padding-right: 6px;
}

.fk-properties {
    height: 100%;
    overflow: auto;
    min-width: 300px;
    max-width: 660px;
    flex: 2 1 auto;
}

.fk-properties__column {
    height: 100%;
    display: flex;
    flex-direction: column;
    overflow: hidden;
}
.fk-properties-label-row {
    min-height: 1.2em;
    display: flex;
    flex-direction: row;
    align-items: baseline;
    justify-content: space-between;
    margin-bottom: 4px;
}
.fk-properties-label-row__left {
    display: flex;
    justify-content: flex-start;
}
.fk-properties__label {
    padding-left: 4px;
    margin-right: 24px;
}
.fk-plots__toolbar {
    margin-bottom: 4px;
    margin-right: 2px;
}
.fk-properties__table {
    flex: 1 1 auto;
    position: relative;
    min-height: 120px;
    max-width: 100%;
}
.fk-controls {
    height: 100%;
    display: flex;
    flex-direction: row;
    padding-top: 8px;
}
.fk-control__grid {
    display: grid;
    grid-template-columns: auto auto;
    justify-content: space-between;
    grid-column-gap: 8px;
    grid-row-gap: 4px;
    align-items: baseline;
}
.fk-control__button {
    margin-top: 24px;
}
.fk {
    display: flex;
    flex: 0 0 auto;
    flex-direction: column;
    width: 300px;
    height: 300px;
    margin-right: 1rem;
}
.fk-rendering {
    position: relative;
    width: 100%;
    height: 100%;
    margin-right: 0.5rem;
    margin-bottom: 0.5rem;
}
.fk-rendering__y-axis {
    position: absolute;
    top: -10px;
    left: 0px;
    bottom: 25px;
    width: 35px;
}
.fk-rendering__x-axis {
    position: absolute;
    right: -10px;
    left: 25px;
    bottom: 0px;
    height: 35px;
}
.fk-rendering__slowness {
    position: absolute;
    left: 35px;
    right: 0px;
    bottom: 0px;
    font-size: 0.5rem;
    text-align: center;
    z-index: 1;
}
.fk-rendering__slowness-label {
    height: calc(100px + 0.5rem);
    width: 260px;
    padding: 0.25rem;
}
.fk-legend {
    height: 50px;
}
.fk-legend__canvas {
    position: absolute;
    right: 100px;
    left: 100px;
    top: 0px;
    z-index: 0;
}
.fk-legend-font__analyst {
    position: absolute;
    left: 120px;
    font-size: $pt-font-size-small;
}
.fk-legend-font__peak {
    position: absolute;
    left: 120px;
    top: 20px;
    font-size: $pt-font-size-small;
}
.fk-legend-font__predicted {
    position: absolute;
    left: 120px;
    top: 36px;
    font-size: $pt-font-size-small;
}
.fk-rendering__canvas {
    position: absolute;
    right: 0px;
    top: 0px;
    z-index: 0;
}
.fk-plot-controls {
    display: inline-flex;
    flex-direction: row;
    align-items: baseline;
    flex: 0 0 auto;
    padding-bottom: 0.25rem;
}
.fk-plot-window-select-control { 
    display: flex;
    flex-direction: row;
    align-items: baseline;
    flex: 0 1 auto;
    padding-right: 6px;
}
.fk-plot-window-control__label {
    padding-right: 4px;
}
.fk-plot-window-control__label-padding {
    padding-left: 6px;
}
.fk-plot-window-numeric-control {
    display: flex;
    flex-direction: row;
    align-items: baseline;
    flex: 0 1 auto;
    padding-left: 6px;
    padding-right: 6px;
    width: 140px;
}
.fk-weavess-wrapper-wrapper {
    flex: 0 1 auto;
    position: relative;
    height: 330px;
}
.fk-channel-list {
    padding: 1rem;
    overflow-y: auto;
    max-height: 200px;
}
.fk-plots-wrapper-1 {
    width: 100%;
    height: 100%;
    color: $gray4;
}
.fk-plots-wrapper-2 {
    height: 100%;
    width: 100%;
    display: flex;
    min-width: fit-content;
    min-height: 332px;
    flex-direction: column;
}
.fk-image-and-details-container {
    padding: 0.5rem;
    height: 100%;
    width: 100%;
    display: flex;
    flex-direction: row;
    user-select: none;
    flex-wrap: wrap;
}
.fk-color-scale {
    position: relative;
    width: 100%;
    height: 100%;
}
.fk-color-scale__xaxis {
    position: absolute;
    right: 0px;
    left: 0px;
    bottom: 0px;
    height: 35px;
    top: 18px
}
.fk-color-scale__canvas {
    position: absolute;
    right: 25px;
    left: 25px;
    top: 0px;
    z-index: 0;
}
.fk-color-scale__units {
    margin-top: 2px;
    position: absolute;
    right: 0px;
    font-size: 11px;
}
.fk-color-scale__button:hover {
    padding: 1px 4px 1px 4px;
    cursor: pointer;
    background-color: white;
    color: black;
    border-radius: 4px;
}

.fk-units-picker {
    display: flex;
}
.fk-units-picker__label {
    margin-right: 8px;
}
.fk-units-picker__radio-group {
    display: flex;
}

.continous-fk-popover {
    padding: 8px 12px 8px 12px;
}

.continous-fk-popover-panel {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
}

.popover-section {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    width: 400px;
    margin-bottom: 8px;

}
.popover-header {
    display: flex;
    justify-content: space-between;
}
.popover-header__text {
    font-size: 14px;
    font-weight: bold;
    line-height: 28px;
}

.popover-body-text {
    font-size: 14px;
    color: rgb(167, 182, 194);
}

.popover-button-group {
    display: flex;
    justify-content: center;
    margin-bottom: 8px;
    margin-top: 4px;
}
.continous-fk-popover-apply-cancel-row {
    display: flex;
    width: 100%;
    justify-content: flex-end;
}

.continous-fk-popover-apply {
    margin-right: 8px;
}

.popover-header__widget {
    display: flex;
    align-items: center;
}

.fk-frequency-thumbnails{
    display: flex;
    flex-wrap: wrap;
    width: 229px;
    padding: .44rem;
}

.fk-properties__frequency-low-high-inputs {
    width: 100px;
}