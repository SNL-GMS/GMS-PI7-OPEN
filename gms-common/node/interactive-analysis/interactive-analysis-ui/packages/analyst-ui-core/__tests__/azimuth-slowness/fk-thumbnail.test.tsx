import { shallow } from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { FkThumbnail, FkThumbnailProps } from '~analyst-ui/components/azimuth-slowness/components/fk-thumbnail';
import { FkUnits } from '~analyst-ui/components/azimuth-slowness/types';

const pixelSize: any = 200;
const mockCallBack = jest.fn();

const fkThumbProps: FkThumbnailProps = {
  // TODO useless test
  fkData: undefined, // This used to be a signal detection (without channel segments) - fk data is undefined
  predictedPoint: undefined,
  sizePx: pixelSize,
  label: 'USRK P',
  selected: false,
  fkUnit: FkUnits.FSTAT,
  dimFk: false,
  onClick: mockCallBack,
  showFkThumbnailMenu: () => {
    /** empty */
  },
};

it('FkThumbnails renders & matches snapshot', () => {
  const tree = renderer.create(<FkThumbnail {...fkThumbProps} />)
    .toJSON();

  expect(tree)
    .toMatchSnapshot();
});

it('FkThumbnails onClick fires correctly', () => {
  const thumbnail = shallow(
    <FkThumbnail
      {
      ...fkThumbProps
      }
    />
  );
  thumbnail.find('.fk-thumbnail')
  .simulate('click');
  expect(mockCallBack)
  .toBeCalled();
});
