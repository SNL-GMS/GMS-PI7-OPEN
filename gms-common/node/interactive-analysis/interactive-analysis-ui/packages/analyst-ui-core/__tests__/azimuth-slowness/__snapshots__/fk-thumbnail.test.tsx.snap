// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FkThumbnails renders & matches snapshot 1`] = `
<div
  className="fk-thumbnail "
  onClick={[MockFunction]}
  onContextMenu={[Function]}
  style={
    Object {
      "height": "200px",
      "width": "200px",
    }
  }
>
  <div
    className="fk-thumbnail__label--reviewed"
  >
    USRK P
  </div>
  <canvas
    className="fk-thumbnail__canvas"
    height={200}
    width={200}
  />
</div>
`;
